src/internal/small_vec.rs

```diff
@@ -108,6 +109,13 @@ impl<T: fmt::Debug> fmt::Debug for SmallVec<T> {
     }
 }
 
+impl<T: Hash> Hash for SmallVec<T> {
```

Comment:
Same here, it looks correct, but why do we need it?

Reply:
This enables `Hash`ing `Ranges` which was something I found very useful in my other project.

Reply:
Ok, sounds useful.